### PR TITLE
Feature/transactions

### PR DESCRIPTION
**Pull request by FredrikAugust**
Added a page for transactions. Complete with options for customizing a new transaction, as well as recurring ones.

### CODE REVIEW

**Comment by FredrikAugust** [comment]
Ready for review. Missing tests, which I will fix in the next PR.

File: src/App.tsx [hunk 1]
```diff
@@ -40,6 +42,15 @@ export default styled(App)`
...
+  nav {
+    overflow-y: hidden;
+  }
+
+  &>section {
+    overflow-y: auto;
+  }
 
   /* Colors */
   background-color: ${props => props.theme.main};
```
Comment (nicolajn) [inline src/App.tsx]: @FredrikAugust Lift this to GlobalStyle?

File: src/components/atoms/Checkbox.tsx [hunk 2]
```diff
@@ -0,0 +1,57 @@
...
+
+interface IProps {
+  className?: string;
+  id: string;
+  setState: React.Dispatch<React.SetStateAction<boolean>>;
+  value: boolean;
+}
+
+const Checkbox: React.FC<IProps> = props => {
+  const updateState = (e: React.FormEvent<HTMLInputElement>) => props.setState(e.currentTarget.checked);
```
Comment (nicolajn) [inline src/components/atoms/Checkbox.tsx]: @FredrikAugust I prefer a different naming convention for event handlers. In stead of passing down a prop called setState, call it onChange, onClick etc.

What are your thinking?

File: src/components/atoms/Checkbox.tsx [hunk 3]
```diff
@@ -0,0 +1,57 @@
+import React from 'react';
+
+import styled from 'styled-components';
+
+interface IProps {
+  className?: string;
+  id: string;
+  setState: React.Dispatch<React.SetStateAction<boolean>>;
```
Comment (nicolajn) [inline src/components/atoms/Checkbox.tsx]: @FredrikAugust I prefer a different naming convention for event handlers. In stead of passing down a prop called setState, call it onChange, onClick etc.

What are your thinking?

File: src/components/atoms/NavigationPill.tsx [hunk 4]
```diff
@@ -27,7 +25,7 @@ export default styled(NavigationPill)`
   font-family: "Open Sans", sans-serif;
```
Comment (nicolajn) [inline src/components/atoms/NavigationPill.tsx]: Font familiy is set in GlobalStyle and don't need to be set here.

File: src/components/atoms/RecurringTransactionOptions.tsx [hunk 5]
```diff
@@ -0,0 +1,48 @@
+import React from 'react';
+
+import Input from './Input';
+
+interface IProps {
```
Comment (nicolajn) [inline src/components/atoms/RecurringTransactionOptions.tsx]: I see we're using different conventions for naming interfaces. I have named mine IComponentNameProps.

File: src/components/atoms/Checkbox.tsx (hunk 3 above)
Comment (nicolajn) [reply src/components/atoms/Checkbox.tsx]: Actually, this is not an event handler, so never mind. A more descriptive prop name is on my wish list.

File: src/components/atoms/NavigationPill.tsx (hunk 4 above)
Comment (FredrikAugust) [reply src/components/atoms/NavigationPill.tsx]: Fixed in local branch. Will push with another PR.

**Pull request merged**